regexp.h

```diff
@@ -170,10 +181,9 @@ typedef struct regexp {
      */
 
     U32 lastparen;           /* highest close paren matched ($+) */
+    U32 lastcloseparen;      /* last close paren matched ($^N) */
     regexp_paren_pair *offs; /* Array of offsets for (@-) and (@+) */
     char **recurse_locinput; /* used to detect infinite recursion, XXX: move to internal */
-    U32 lastcloseparen;      /* last close paren matched ($^N) */
-
 
```

Comment:
This makes struct regexp 8 bytes larger on 8 byte alignment platforms.

Whether the smaller size is worth the loss of readability is a matter of taste, you can ignore this comment if you wish.

Reply:
Oh, I think I miscounted then. I'll revert that change, I had the opposite intent. Good catch.

Reply:
It doesn't require much counting:
- `void *pprivate;` is 8-byte aligned,
- the `U32 intflags;` and `U32 lastparen;` take us to the next 8 byte boundary
- so adding another U32 here means we need padding before `offs`
- we have only the two bitfields after `recurse_input` which should combine to a single U32
- so without `lastcloseparen` we need another 4 bytes of padding to 8 byte align `subbeg`.

The change you've made instead (moving `extflags` as well as `lastcloseparan` looks good.

Reply:
Agreed. I just wondered if you had some "trick" to make this easier. I actually started wondering if we should have a test which shows the size of the structs.

Reply:
A little research found `pahole`:
```
$ pahole -C regexp ./perl
struct regexp {
        HV *                       xmg_stash;            /*     0     8 */
        union _xmgu                xmg_u;                /*     8     8 */
        STRLEN                     xpv_cur;              /*    16     8 */
...
        /* size: 224, cachelines: 4, members: 32 */
        /* sum members: 216, holes: 1, sum holes: 4 */
        /* sum bitfield members: 13 bits, bit holes: 1, sum bit holes: 19 bits */
        /* last cacheline: 32 bytes */
};
$ pahole -P ./perl
pmop    96      88      8
custom_op       40      32      8
block_sub       48      40      8
RExC_state_t    416     408     8
_IO_FILE        216     208     8
```
I think this is the tool Richard used

Reply:
Yeah, that's what I've used previously. Was only manually eyeballing though, I do not have any code around it.